### PR TITLE
✨ Store computational services progress in comp_tasks table, decouple webserver from api-server 🗃️

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- [x] add *progress* column to **comp_tasks** table
- [x] adapt director-v2 to store last progress in **comp_tasks** table
- [x] adapt director-v2 to return the progress from **comp_tasks** table
- [x] adapt director-v2 to return overall pipeline progress (based on pipeline each node progress)
- [x] dask-sidecar: uncontrolled progress value from comp. services gets forcibly set between 0 and 1
- [x] api-server now returns progress as well (https://github.com/ITISFoundation/osparc-simcore/issues/2444)

NOTE: Next PR should then remove the progress changes in the DB done by the webserver. This should completely decouple the webserver from the api-server run pipelines.

Bonus:
- [x] fix flaky with migration service (remove service instead of killing the task which would trigger a double restart)
- [x] some sqlalchemy 2.0 refactoring (use ```Row._asdict()``` to convert to a dict)

## Related issue/s
- resolves https://github.com/ITISFoundation/osparc-simcore/issues/2444
- relates to #3991 
- relates to https://github.com/ITISFoundation/osparc-issues/issues/675
- relates to https://github.com/ITISFoundation/osparc-simcore/issues/3820
- relates to https://github.com/ITISFoundation/osparc-simcore/issues/3756

### CODE REVIEW

File: services/director-v2/setup.cfg [hunk 1]
```diff
@@ -11,3 +11,4 @@ commit_args = --no-verify
 asyncio_mode = auto
 markers =
 	testit: "marks test to run during development"
+  acceptance_test: "marks tests as 'acceptance tests' i.e. does the system do what the user expects? Typically those are workflows."
```
Comment (pcrespov) [inline services/director-v2/setup.cfg:14]: careful here with tabs and spaces. ! setup will not understand the marker if you did it wrong

I tried to setup vscode to format *.cfg with tabs/spaces but did not manage. If you find a way, please add it to the .vscode/settings.template.json

File: tests/public-api/test_solvers_jobs_api.py [hunk 2]
```diff
@@ -224,12 +224,12 @@ def test_run_job(
         status: JobStatus = solvers_api.inspect_job(solver.id, solver.version, job.id)
         assert isinstance(status, JobStatus)
 
-        assert 0 <= status.progress <= 100
+        assert 0 <= status.progress <= 1
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply tests/public-api/test_solvers_jobs_api.py:227]: no it's sadly still 100. I will fix it. good catch

[the remark this answers is not in the document]
Comment (sanderegg) [reply tests/public-api/test_solvers_jobs_api.py:227]: but the test does not catch it anyway.

File: services/director-v2/setup.cfg (hunk 1 above)
Comment (sanderegg) [reply services/director-v2/setup.cfg:14]: ok.

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Review by GitHK — APPROVED** [review]

File: services/director-v2/tests/integration/conftest.py [hunk 3]
```diff
@@ -109,3 +115,47 @@ def mock_projects_repository(mocker: MockerFixture) -> None:
         f"{module_base}.ProjectsRepository.is_node_present_in_workbench",
         return_value=mocked_obj,
     )
+
+
+@pytest.fixture
+async def catalog_ready(
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/director-v2/tests/integration/conftest.py:121]: well it's only used here at the moment.

File: services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py [hunk 4]
```diff
@@ -244,23 +245,22 @@ async def upsert_tasks_from_project(
...
-                t.node_id
-                for t in await result.fetchall()
-                if t.node_id not in project.workbench
-            ]
-            for node_id in node_ids_to_delete:
-                await conn.execute(
-                    sa.delete(comp_tasks).where(
-                        (comp_tasks.c.project_id == str(project.uuid))
-                        & (comp_tasks.c.node_id == node_id)
+            if all_nodes := await result.fetchall():
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_tasks.py:253]: yep

**Pull request merged**